spanner/api/Program.cs

```diff
@@ -0,0 +1,965 @@
+                            secondBudget =
+                                decimal.Parse(reader["MarketingBudget"]
+                                    .ToString());
+                            // Confirm second Album has > 300,000 and if not
+                            // raise an exception. Raising an exception will 
+                            // automatically roll back the transaction.
+                            if (secondBudget < minimumAmountToTransfer)
+                            {
+                                throw new Exception("The second album's "
+                                    + $"budget {secondBudget} "
+                                    + "is less than the minimum required "
+                                    + "amount to transfer.");
+                            }
+                        }
+                    }
+                    // Read the first album's budget.
+                    cmdLookup = connection.CreateSelectCommand(
+                    "SELECT * FROM Albums WHERE SingerId = 1 and AlbumId = 1");
+                    using (var reader = await cmdLookup.ExecuteReaderAsync())
+                    {
+                        while (await reader.ReadAsync())
+                        {
+                            firstBudget =
+                                decimal.Parse(reader["MarketingBudget"]
+                                    .ToString());
+                        }
+                    }
+
+                    // Specify update command parameters.
+                    var cmd = connection.CreateUpdateCommand("Albums",
+                        new SpannerParameterCollection {
+                        {"SingerId", SpannerDbType.Int64},
+                        {"AlbumId", SpannerDbType.Int64},
+                        //{"AlbumTitle", SpannerDbType.String},
+                        {"MarketingBudget", SpannerDbType.Int64},
+                    });
+                    // Update second album to remove the transfer amount.
+                    secondBudget = secondBudget - transferAmount;
+                    cmd.Parameters["SingerId"].Value = 2;
+                    cmd.Parameters["AlbumId"].Value = 2;
+                    cmd.Parameters["MarketingBudget"].Value = secondBudget;
+                    await cmd.ExecuteNonQueryAsync();
+                    // Update first album to add the transfer amount.
+                    firstBudget = firstBudget + transferAmount;
+                    cmd.Parameters["SingerId"].Value = 1;
+                    cmd.Parameters["AlbumId"].Value = 1;
+                    cmd.Parameters["MarketingBudget"].Value = firstBudget;
+                    await cmd.ExecuteNonQueryAsync();
+
+                    // Commit the transaction to Spanner table.
+                    scope.Complete();
+                    // Yield Task thread back to the current context.
+                    await Task.Yield();
+                    Console.WriteLine("Transaction complete.");

```

Comment:
you should be using topaz as well here to handle retries:

https://www.codeproject.com/Articles/769634/Transient-Fault-Handling-Application-Block-Simplif

you can use the extension method exception.IsTransientSpannerFault() for your implementation of a detection strategy.
let me know if you have more questions about this and I can go over it in more detail.